### ISSUE TITLE
Cursor jumps to time in date time picker

### ISSUE THREAD

**Issue by gerard-isdell** [cerner/terra-framework#1168]
# Bug Report

## Description
When using select date/ time button and typing in a date, the cursor jumps from month to time or day to time. This is if drop your cursor in the field without clearing the whole date.

https://engineering.cerner.com/terra-ui/components/terra-date-time-picker/date-time-picker/date-time-picker  

## Steps to Reproduce
1. Open https://engineering.cerner.com/terra-ui/components/terra-date-time-picker/date-time-picker/date-time-picker
2. Highlight the date or month
3. Enter a new date or month
4. The cursor jumps to the time field

## Additional Context / Screenshots

https://cernerprod.sharepoint.com/sites/TerraTimelineIncubator/Shared%20Documents/General/Screen%20Recording%202020-06-30%20at%2008.07.17.mov

## Expected Behavior
After you entered the new date or month, the cursor should remain where it is

## Environment

* Date Time Picker 4.4.40
* Browser Name and Version: Safari
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile):

**Comment by benbcai** [comment]
@gerard-isdell There is logic that automatically moves the focus to the time input once the date input has a valid date. If the date input does not have a complete and valid date, the focus would stay in the date input. The purpose of this is to guide the user to enter the date followed by the time without having to manually move.

This is similar to how the time input would jump the focus from the hour input to the minute input once the hour is completed.

**Comment by ryanthemanuel** [comment]
@gerard-isdell we do want the cursor to jump as you're entering the field as a convenience; however, right now I do know that things get a little weird with an existing date and updating the month (it'll jump to the time at that point rather than allowing you to update the date as well). We believe this scenario will be fixed when we address https://github.com/cerner/terra-framework/issues/468 which is currently being worked on. It will then jump from field to field automatically (month to date to time). Will that address your concerns?

cc @neilpfeiffer @benbcai

**Comment by neilpfeiffer** [comment]
Thanks @gerard-isdell, as @benbcai commented, this is technically performing as designed, but based on your video, I can see why this isn't performing as you might expect. 

The current date-picker uses only a single input, so once a fully complete and valid date is detected, in the combined date-time-picker it auto-jumps to the time-input. Which appears to work fine for an empty date input, that only after a full month, day, and year are added it is complete, with year being last to then jump to the time-input. But is odd when only adjusting a month or day in an already complete date. 

We have plans to update this internal construction of the date-input section to 3 distinct inputs: one for month, one for day, and one for year -- https://github.com/cerner/terra-framework/issues/468  -- which our time-input is currently constructed in this way and advances between the hours to minutes to seconds. After the separation, we would expect the interaction in your video to change, that after completing a valid month to auto-advance to the day input (assuming for month/day/year localized date format).

**Comment by gerard-isdell** [comment]
Thanks @neilpfeiffer @benbcai - that all sounds fine.

**Comment by ryanthemanuel** [comment]
@gerard-isdell if that's the case, I'll close this as a duplicate of #468
